public/app/plugins/datasource/tempo/SearchTraceQLEditor/SearchField.tsx

```diff
@@ -112,13 +113,25 @@ const SearchField = ({
       operatorList = numberOperators;
   }
 
+  /**
+   * Add to a list of options the current template variables.
+   *
+   * @param options a list of options
+   * @returns the list of given options plus the template variables
+   */
+  const withTemplateVariableOptions = (options: SelectableValue[] | undefined) => {
+    const templateSrv: TemplateSrv | undefined = getTemplateSrv(); // `undefined`in several tests
```

Comment:
The TemplateSrv has to be set in the test with `setTemplateSrv` to some mock service. That way we can remove the `undefined` here and fix the tests.